### DISCUSSION TITLE
useMatches doesn't seem to work properly with index routes nested under directories

### DISCUSSION THREAD

**Discussion by illegalnumbers** [remix-run/remix#5082] [Q&A]
I have a route structure laid out like the following:
```
routes
  foo/
    - index 
    - bar/
        - index
 - root
```

When I use `useMatches` in my root to lay out a breadcrumb I have a `handle` defined in both `foo/index` and `bar/index`. I expect the matches when I'm at `/foo/bar` to include the `foo` handle with my breadcrumb for that route. It looks though that I only get the `bar` handle displayed without the original index.

If I change it so that `foo/index` is at the root and called `foo` it renders my `foo/index` component but the breadcrumb looks correct when I'm at the `foo/bar` route.

Is there something I'm doing wrong here?

**Comment by danreeves** [comment]
Having the same problem. My current solution is to have intermediate "pages" that just render an outlet

```
routes/
	my-route.tsx # Just an Outlet
	my-route/
		index.tsx # List page
		$sub-route/
			index.tsx # Individual page
```

This allows me to see both my-route and $sub-route in the useMatches array to construct breadcrumbs.

If  I remove `my-route.tsx` it disappears from the useMatches array when I'm at `/my-route/sub-route`
